src/components/pages/OrderStatus/OrderStatus.js

```diff
@@ -2,32 +2,153 @@ import React from 'react';
+  static propTypes = {
     actions: PropTypes.object.isRequired,
-    fuelSavings: PropTypes.object.isRequired
-  };
+    selectedShoppingItems: PropTypes.array.isRequired,
+    dailyCurrentPrices: PropTypes.object.isRequired,
+    updatePurchasePricesForDay: PropTypes.func.isRequired,
+  }
 
-  saveFuelSavings = () => {
-    this.props.actions.saveFuelSavings(this.props.fuelSavings);
-  };
+  state = {
+    day: 0,
+  }
 
-  calculateFuelSavings = e => {
-    this.props.actions.calculateFuelSavings(this.props.fuelSavings, e.target.name, e.target.value);
-  };
+  onResetClick = () => {
+    const { resetPurchasePrices } = this.props.actions;
+
+    this.setState({ day: 0 });
+    resetPurchasePrices();
+  }
+
+  onNextDayClick = () => {
+    const { updatePurchasePricesForDay } = this.props.actions;
+    const nextDay = this.state.day + 1;
+
+    this.setState(() => ({ day: nextDay }));
+    updatePurchasePricesForDay({ day: nextDay });
+  }
+
+  renderItems = () => {
+    const { selectedShoppingItems, dailyCurrentPrices } = this.props;
+    const currentDay = this.state.day;
+
+    return selectedShoppingItems.map((item) => {
+      const { name, category, purchasePrice, goodUntil } = item;
+      const currentPrice = dailyCurrentPrices[category][currentDay];
+      const dynamicGoodUntil = Math.max(0, goodUntil - this.state.day);
+
+      return (
+        <div style={ styles.itemRow } key={ category }>
+          <div style={ styles.itemValue }>{ name }</div>
+          <div style={ styles.itemValue }>$&nbsp;{ purchasePrice || '-' }</div>
+          <div style={ styles.itemValue }>$&nbsp;{ currentPrice }</div>
+          <div style={ styles.itemValue }>{ dynamicGoodUntil > 0 ? `${ dynamicGoodUntil } day(s)` : 'Not checking anymore' }</div>
+        </div>
+      );
+    });
+  }
 
   render() {
-    return (<div>hello</div>
+    const { dailyCurrentPrices, selectedShoppingItems } = this.props;
+    const nextDayButtonStatus = this.state.day === dailyCurrentPrices.toilet_paper.length - 1 ? 'disabled' : '';
+    const estimatedPrice = selectedShoppingItems.reduce((acc, item) => (acc + (item.quantity * item.purchasePrice)), 0);
+    const totalHoneyGold = selectedShoppingItems.reduce((acc, item) => (acc + (item.quantity * item.honeyGold)), 0);
+    const totalSavings = totalHoneyGold / 100;
+
+    return (
+      <div style={ styles.main }>
+        <div style={ styles.buttons }>
+          <div style={ styles.resetButton}><Button buttonType="secondary-ghost" copy="Reset" onClick={ this.onResetClick } /></div>
+          <Button buttonType="secondary-ghost" copy="Next Day" onClick={ this.onNextDayClick } status={ nextDayButtonStatus } />
+        </div>
+        <div style={ styles.titleSection }>
+          <div style={ styles.titleValue }>Product</div>
+          <div style={ styles.titleValue }>Purchase Price</div>
+          <div style={ styles.titleValue }>Current Price</div>
+          <div style={ styles.titleValue }>Good Until</div>
```

Comment:
Could we rename `Good Until` to `Expiration Date`? I think that would make more sense to users